library.js

```
@@ -198,11 +198,17 @@ plugin.normalizePayload = function(payload, callback) {
 	});
 };
 
-plugin.verifyUser = function(uid, callback) {
-	// Check ban state of user, reject if banned
-	user.isBanned(uid, function(err, banned) {
-		callback(err || banned ? new Error('banned') : null, uid);
-	});
+plugin.verifyUser = function (token, uid, isNewUser, callback) {
+    plugins.fireHook('static:sessionSharing.verifyUser', {
+    	uid: uid,
+    	isNewUser: isNewUser,
+    	token: token
+    }, function (err) {
```

A plugin could pass its own error back (in fact, I'd imagine that's the whole point), in which case you can actually return the error early instead of checking ban state.

---

Good point! Added in the error check